src/core/algorithms/association_rules/ar_verifier/ar_stats_calculator.cpp

```diff
@@ -0,0 +1,98 @@
+#include "ar_stats_calculator.h"
+
+#include <queue>
+#include <set>
+
+#include <easylogging++.h>
+
+namespace algos::ar_verifier {
+double ARStatsCalculator::CalculateTransactionCoverage(
+        std::vector<unsigned> const& transaction_indices, std::vector<unsigned> const& rule_part) {
+    if (transaction_indices.empty()) {
+        return 0.0;
+    }
+
+    if (rule_part.empty()) {
+        return 1.0;
+    }
+
+    std::set transaction_set(transaction_indices.begin(), transaction_indices.end());
+    std::set rule_set(rule_part.begin(), rule_part.end());
+
+    std::vector<unsigned> intersection;
+    std::ranges::set_intersection(rule_set, transaction_set, std::back_inserter(intersection));
+
+    return intersection.size() / static_cast<double>(rule_set.size());
+}
+
+ClusterPriority ARStatsCalculator::CalculateClusterPriority(
+        std::pair<double, double> const& coverage) {
+    bool const is_left_full = coverage.first == 1.0;
+    bool const is_right_full = coverage.second == 1.0;
+    bool const is_right_present = coverage.second > 0.0;
+
+    if (is_left_full) {
+        if (is_right_present) {
+            if (is_right_full) return ClusterPriority::full_left_full_right;
+            return ClusterPriority::full_left_partial_right;
+        }
+        return ClusterPriority::full_left_no_right;
+    }
+    if (is_right_present) {
+        if (is_right_full) return ClusterPriority::partial_left_full_right;
+        return ClusterPriority::partial_left_partial_right;
+    }
+    return ClusterPriority::partial_left_no_right;
+}
+
+void ARStatsCalculator::CalculateRuleCoverageCoefficients() {
+    for (auto const& [id, itemset] : data_->GetTransactions()) {
+        double coverage_left = CalculateTransactionCoverage(itemset.GetItemsIDs(), rule_.left);
+        double coverage_right = CalculateTransactionCoverage(itemset.GetItemsIDs(), rule_.right);
```

Comment:
nit: I see that you use `const` for local variables. But in some places (like this one) you don't write them. I suggest adding them in all places for consistency here and in similar places. (This is not necessary)